### PR TITLE
Fixed Window.position for Winforms

### PR DESCRIPTION
**Pull request by t-arn**
This PR fixes the problem for winforms that the window position could neither be set nor queried.
It contains following changes:
* implemented set position in Window constructor for winforms
* implemented set position by setting Window.position for winforms
* implemented get position by getting Window.position for winforms
* implemented Window.get_position in toga_dummy/window.py
* added not_implemented('Window.get_position()') for all other platforms

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

**Comment by t-arn** [comment]
I don't understand what the "styling issues" are and what I should do to solve these issues.
Also, I'm relatively new to Python and completely new to BeeWare - so, I don't know how to write the unit tests.

**Comment by t-arn** [comment]
OK, now I found the details about the problem - I didn't realize that I can drill down into the test results

**Comment by t-arn** [comment]
I just realized that I on the not_implemented platforms, I **need** to return something or the position will become None!
I'll fix this in the next commit.

File: src/core/toga/window.py [hunk 1]
```diff
@@ -158,6 +158,7 @@ def position(self):
         Returns:
             A ``tuple`` of (``int``, ``int``) int the from (x, y).
         """
+        self._position = self._impl.get_position()
```
Comment (freakboy3742) [inline src/core/toga/window.py:161]: There's no need to cache `self._position` if the backend is responsible for providing a literal value.

File: src/winforms/toga_winforms/window.py [hunk 2]
```diff
@@ -44,6 +44,8 @@ def create(self):
         self.native.Resize += self.winforms_resize
         self.toolbar_native = None
         self.toolbar_items = None
+        self.native.StartPosition = 0   # FormStartPosition.Manual
```
Comment (freakboy3742) [inline src/winforms/toga_winforms/window.py:47]: What exactly does this statement do? Setting Location makes sense; but it isn't clear why "StartPosition = 0" is needed (and the comment at the end of the line doesn't make much sense either). A clarifying comment about exactly what is needed would be helpful here, if only to explain to future explorers what is going on.

File: src/android/toga_android/window.py [hunk 3]
```diff
@@ -46,6 +46,10 @@ def set_content(self, widget):
     def set_title(self, title):
         pass
 
+    def get_position(self):
+        self.interface.factory.not_implemented('Window.get_position()')
```
Comment (saroad2) [inline src/android/toga_android/window.py:50]: If you already implemented the method, no need for that warning :)
The same goes for the other platforms.

File: src/core/toga/window.py (hunk 1 above)
Comment (t-arn) [reply src/core/toga/window.py:161]: yes, I think it is. What would you then return from the platforms where we do not yet have an implementation? In the current code base, self._position **does** seem to have a value and it currently **is** this value that is returned by the position getter.

File: src/winforms/toga_winforms/window.py (hunk 2 above)
Comment (t-arn) [reply src/winforms/toga_winforms/window.py:47]: In Winforms, FormStartPosition is an enum which defines how the new window should be positioned: https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.formstartposition?view=netcore-3.1

Other values would be 4 (center on parent) or 1 (center on screen). Setting it to 0 allows you to set the position by x/y coordinates

Comment (freakboy3742) [reply src/winforms/toga_winforms/window.py:47]: Ah - if FormStartPosition is an enumeration, then we should be using that symbol directly from Winforms. You should find `WinForms.FormStartPosition` is available as a symbol directly.

File: src/core/toga/window.py (hunk 1 above)
Comment (freakboy3742) [reply src/core/toga/window.py:161]: If the backend doesn't have an implementation, it's *not implemented*.  We shouldn't be baking in features to the stable frontend interface to compensate for an unstable backend.

File: src/winforms/toga_winforms/window.py (hunk 2 above)
Comment (t-arn) [reply src/winforms/toga_winforms/window.py]: OK, used WinForms.FormStartPosition.Manual instead of 0 and added an explaining comment

**Comment by t-arn** [comment]
I now made the implementation the source of truth for the window position on all platforms. What remains to be done is actually getting the correct position from the platforms where this is not yet done.

File: src/winforms/toga_winforms/window.py [hunk 4]
```diff
@@ -45,6 +45,10 @@ def create(self):
         self.toolbar_native = None
         self.toolbar_items = None
 
+        # Tell Winforms to heed self.native.Location on window creation
+        self.native.StartPosition = WinForms.FormStartPosition.Manual
+        self.native.Location = Point(*self.interface.position)
```
Comment (freakboy3742) [inline src/winforms/toga_winforms/window.py:50]: `self.interface.position` is determined by invoking `get_position` - so you'll be setting the initial position of the window to the position that window is currently in. The position provided by the user at the time of construction will be lost. This is a bootstrapping problem.

**Review by freakboy3742 — CHANGES_REQUESTED** [review]

**Comment by t-arn** [comment]
Deleted the branch by mistake which closed this PR.
Restored the branch and reopened this PR.

File: src/core/toga/window.py (hunk 1 above)
Comment (t-arn) [reply src/core/toga/window.py:161]: OK, removed self._position variable

File: src/android/toga_android/window.py [hunk 5]
```diff
@@ -46,8 +46,11 @@ def set_content(self, widget):
     def set_title(self, title):
         pass
 
+    def get_position(self):
+        return self._position
+
     def set_position(self, position):
-        pass
+        self._position = (0, 0)  # windows are always full screen in Android
```
[the remark this answers is not in the document]
Comment (t-arn) [reply src/android/toga_android/window.py:53]: OK, removed self._position

File: src/cocoa/toga_cocoa/window.py [hunk 6]
```diff
@@ -216,8 +216,13 @@ def set_content(self, widget):
...
         self.native.title = title
 
+    def get_position(self):
+        self.interface.factory.not_implemented('Window.get_position()')
+        return self._position
+
     def set_position(self, position):
-        pass
+        self.interface.factory.not_implemented('Window.set_position()')
+        self._position = position
```
[the remark this answers is not in the document]
Comment (t-arn) [reply src/cocoa/toga_cocoa/window.py:225]: OK, removed self._position

File: src/gtk/toga_gtk/window.py [hunk 7]
```diff
@@ -132,8 +132,13 @@ def on_size_allocate(self, widget, allocation):
...
         self.native.close()
 
+    def get_position(self):
+        self.interface.factory.not_implemented('Window.get_position()')
+        return self._position
+
     def set_position(self, position):
-        pass
+        self.interface.factory.not_implemented('Window.set_position()')
+        self._position = position
```
[the remark this answers is not in the document]
Comment (t-arn) [reply src/gtk/toga_gtk/window.py:141]: OK, removed self._position

File: src/iOS/toga_iOS/window.py [hunk 8]
```diff
@@ -57,8 +57,11 @@ def set_content(self, widget):
     def set_title(self, title):
         pass
 
+    def get_position(self):
+        return self._position
+
     def set_position(self, position):
-        pass
+        self._position = (0, 0)  # windows are always full screen in iOS
```
[the remark this answers is not in the document]
Comment (t-arn) [reply src/iOS/toga_iOS/window.py:64]: OK, fixed

File: src/web/toga_web/window.py [hunk 9]
```diff
@@ -75,8 +75,13 @@ def on_size_allocate(self, widget, allocation):
     def close(self):
         self.interface.factory.not_implemented('Window.close()')
 
+    def get_position(self):
+        self.interface.factory.not_implemented('Window.get_position()')
+        return self._position
+
     def set_position(self, position):
         self.interface.factory.not_implemented('Window.set_position()')
+        self._position = position
```
[the remark this answers is not in the document]
Comment (t-arn) [reply src/web/toga_web/window.py:84]: OK, I now ignore set_position() and return (0, 0) in get_position(), same as on mobile platforms

File: src/winforms/toga_winforms/window.py (hunk 4 above)
Comment (t-arn) [reply src/winforms/toga_winforms/window.py:50]: hmm...I see what you mean, but it **does** work: the position provided from the user is **not** lost and the window appears where it should (see the window.py example)

**Comment by t-arn** [comment]
@saroad2 I still don't know how to add a unit test for this, but I made an example (window.py)

**Comment by t-arn** [comment]
@freakboy3742 Addressed all comments, please re-review

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
Thanks for this contribution! There's an overlap between this PR and #1395; and both PRs are subject to the "source of truth" problem. I've added #1461 to merge the two PRs, adding fixes to address the source of truth issue. As a result, I'm going to close this PR; but it was really helpful in getting that PR working, so thanks for the assist!

**Pull request closed without merging**